src/sentry/models/release.py

```diff
@@ -258,13 +258,10 @@ def set_refs(self, refs, user, fetch=False):
         ).exclude(version=self.version).order_by('-date_added').first()
 
         for ref in refs:
-            try:
-                repo = Repository.objects.get(
-                    organization_id=self.organization_id,
-                    name=ref['repository'],
-                )
-            except Repository.DoesNotExist:
-                continue
+            repo = Repository.objects.get(
+                organization_id=self.organization_id,
+                name=ref['repository'],
+            )
```

Comment:
So there is a possibility here that we'd still raise an exception due to a race condition.

Let's refactor this into something like:


```python
names = {r['repository'] for r in refs}
repos = list(Repository.objects.filter(
  organization_id=self.organization_id,
  name__in=names,
))
# do our validation here to determine if all the names match.
repos_by_name = {r['name']: r for r in repos}
for ref in refs:
    repo = repos_by_name[ref['repository']
    ...
```

This has the benefit of also reducing the query count.